File: program/WordCount/Func/Top10WordList.java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.Arrays;
import java.util.stream.Collectors;
import java.util.Comparator;
import java.util.Map.Entry;

class Top10WordList {
    public static void main(String[] args) throws Exception {
        Files.readAllLines(Paths.get("news.txt"))
        .stream()
        .map(line->line.toLowerCase().replaceAll("[.,!'()-]",""))
        .flatMap(line->Arrays.stream(line.split(" +")))
        .collect(Collectors.groupingBy(x->x,Collectors.counting()))
        .entrySet()
        .stream()
        .sorted(Comparator.comparing(Entry<String,Long>::getValue).reversed()
                .thenComparing(Entry<String,Long>::getKey))
        .limit(10)
        .forEach(x->System.out.printf("%d %s\n",x.getValue(),x.getKey()));
    }
}

File: program/WordCount/Func/Top10WordList.pl
use List::Util (reduce);

open(FILE,"< news.txt");
printf("%d %s\n",$_->[1],$_->[0]) for
grep {$i++<10}
sort {$b->[1]<=>$a->[1] or $a->[0] cmp $b->[0]}
map {@{$_}}
reduce {($n=@{$a})==0?[[$b,1]]:$a->[$n-1]->[0] ne $b ? [@{$a},[$b,1]] : [@{$a}[0..$n-2],[$b,$a->[$n-1]->[1]+1]]}
[], sort {$a cmp $b}
map {chomp;tr/A-Z/a-z/;tr/.,!'()-//d;split}
<FILE>;
close(FILE);
